app/src/main/java/com/android/sample/ui/createRecipe/PublishRecipeScreen.kt

```diff
@@ -0,0 +1,110 @@
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.sample.model.recipe.CreateRecipeViewModel
+import com.android.sample.resources.C.Tag.CHEF_IMAGE_DESCRIPTION
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.theme.Typography
+import com.android.sample.ui.utils.PlateSwipeScaffold
+
+@Composable
+fun PublishRecipeScreen(
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel,
+) {
+  PlateSwipeScaffold(
+      navigationActions = navigationActions,
+      selectedItem = Route.CREATE_RECIPE,
+      showBackArrow = true,
+      content = { paddingValues ->
+        PublishRecipeContent(
+            navigationActions,
+            createRecipeViewModel,
+            modifier = Modifier.padding(paddingValues).fillMaxSize())
+      })
+}
+
+/**
+ * Composable function that displays the content for publishing a recipe.
+ *
+ * @param navigationActions Actions for navigating between screens.
+ * @param createRecipeViewModel ViewModel for managing the recipe creation process.
+ * @param modifier Modifier to be applied to the content.
+ */
+@Composable
+fun PublishRecipeContent(
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel,
+    modifier: Modifier = Modifier
+) {
+  // Get the current context
+  val context = LocalContext.current
+
+  // Collect the publish error state
+  val publishError = createRecipeViewModel.publishError.collectAsState(initial = null).value
+
+  // Show a toast message if there is a publish error
+  publishError?.let {
+    Toast.makeText(context, it, Toast.LENGTH_SHORT).show()
+    createRecipeViewModel.clearPublishError()
+  }
+
+  Column(
+      modifier = modifier,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.SpaceBetween) {
+        Spacer(modifier = Modifier.weight(0.05f))
+        // Display the done text
+        Text(
+            text = stringResource(R.string.done_text),
+            style = Typography.titleLarge.copy(fontSize = 70.sp, fontWeight = FontWeight.Bold),
+            modifier = Modifier.weight(0.4f).padding(bottom = 16.dp).testTag("DoneText"))
+        Spacer(modifier = Modifier.weight(0.1f))
+
+        // Display the chef in egg image
```

Comment:
I love the idea of the chef breaking out of the egg at the end of creating a long recipe, it’s such a fun and satisfying way to celebrate the user’s effort and completion! It really adds a nice touch to the experience.  

Just one small thing: on my phone, the image looks a bit low-res. Is there a way we could enhance the image quality to ensure it’s crisp on all devices? It would make this ending moment even more impactful!


![image](https://github.com/user-attachments/assets/1a5343e0-15c5-4c65-bece-2106bbdf33d8)

Reply:
I dont what could be the issue. Maybe the original ratio doesnt hold anymore? I will try to respect this original ratio

Reply:
I have done it